### ISSUE TITLE
Unable to download macOS x64 of geckodriver 0.29.1

### ISSUE THREAD

**Issue by thc202** [bonigarcia/webdrivermanager#634]
**Description of the problem**:
The WebDriver downloaded does not match the architecture requested, expected x64 but got aarch64.

**WebDriverManager version**:
4.4.1

**WebDriverManager use**:
```Java
WebDriverManager.firefoxdriver()
        .avoidBrowserDetection()
        .avoidReadReleaseFromRepository()
        .avoidFallback()
        .driverVersion("0.29.1")
        .operatingSystem(OperatingSystem.MAC)
        .architecture(Architecture.X64)
        .setup();
```

**WebDriverManager traces**:
```
1169 [main] DEBUG io.github.bonigarcia.wdm.cache.ResolutionCache - Created new resolution cache file at ~/.cache/selenium/resolution.properties
1635 [main] TRACE io.github.bonigarcia.wdm.cache.CacheHandler - Checking if geckodriver exists in cache
1648 [main] TRACE io.github.bonigarcia.wdm.cache.CacheHandler - Filter cache by geckodriver -- input list [~/.cache/selenium/resolution.properties] -- output list [] 
1649 [main] TRACE io.github.bonigarcia.wdm.cache.CacheHandler - Filter cache by 0.29.1 -- input list [] -- output list [] 
1649 [main] TRACE io.github.bonigarcia.wdm.cache.CacheHandler - Filter cache by MAC -- input list [] -- output list [] 
1649 [main] TRACE io.github.bonigarcia.wdm.cache.CacheHandler - Filter cache by 64 -- input list [] -- output list [] 
1649 [main] TRACE io.github.bonigarcia.wdm.cache.CacheHandler - geckodriver not found in cache
1651 [main] INFO  io.github.bonigarcia.wdm.WebDriverManager - Reading https://api.github.com/repos/mozilla/geckodriver/releases to seek geckodriver
2269 [main] TRACE io.github.bonigarcia.wdm.WebDriverManager - All driver URLs: [https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux32.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux32.tar.gz.asc, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux64.tar.gz.asc, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-win32.zip, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-win64.zip, https://github.com/mozilla/geckodriver/releases/download/v0.29.0/geckodriver-v0.29.0-linux32.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.0/geckodriver-v0.29.0-linux32.tar.gz.asc, https://github.com/mozilla/geckodriver/releases/download/v0.29.0/geckodriver-v0.29.0-linux64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.0/geckodriver-v0.29.0-linux64.tar.gz.asc, https://github.com/mozilla/geckodriver/releases/download/v0.29.0/geckodriver-v0.29.0-macos.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.0/geckodriver-v0.29.0-win32.zip, https://github.com/mozilla/geckodriver/releases/download/v0.29.0/geckodriver-v0.29.0-win64.zip,  ...]
2289 [main] DEBUG io.github.bonigarcia.wdm.WebDriverManager - Driver to be downloaded geckodriver 0.29.1
2289 [main] TRACE io.github.bonigarcia.wdm.WebDriverManager - Driver URLs after filtering for version: [https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux32.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux32.tar.gz.asc, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux64.tar.gz.asc, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-win32.zip, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-win64.zip]
2289 [main] TRACE io.github.bonigarcia.wdm.online.UrlHandler - URLs before filtering by OS (MAC): [https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux32.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux32.tar.gz.asc, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-linux64.tar.gz.asc, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-win32.zip, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-win64.zip]
2295 [main] TRACE io.github.bonigarcia.wdm.online.UrlHandler - URLs after filtering by OS (MAC): [https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos.tar.gz]
2295 [main] TRACE io.github.bonigarcia.wdm.online.UrlHandler - URLs before filtering by architecture (64): [https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz, https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos.tar.gz]
2297 [main] TRACE io.github.bonigarcia.wdm.online.UrlHandler - URLs after filtering by architecture (64): [https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz]
2299 [main] TRACE io.github.bonigarcia.wdm.online.UrlHandler - URLs before filtering by beta versions: [https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz]
2299 [main] TRACE io.github.bonigarcia.wdm.online.UrlHandler - URLs after filtering by beta versions: [https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz]
2305 [main] TRACE io.github.bonigarcia.wdm.online.Downloader - Target file for URL https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz driver version 0.29.1 = ~/.cache/selenium/geckodriver/mac64/0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz
2305 [main] INFO  io.github.bonigarcia.wdm.online.Downloader - Downloading https://github.com/mozilla/geckodriver/releases/download/v0.29.1/geckodriver-v0.29.1-macos-aarch64.tar.gz
2307 [main] TRACE io.github.bonigarcia.wdm.online.Downloader - Target folder ~/.cache/selenium/geckodriver/mac64/0.29.1 ... using temporal file /tmp/570237492477157912/geckodriver-v0.29.1-macos-aarch64.tar.gz
2842 [main] INFO  io.github.bonigarcia.wdm.online.Downloader - Extracting driver from compressed file geckodriver-v0.29.1-macos-aarch64.tar.gz
2951 [main] TRACE io.github.bonigarcia.wdm.online.Downloader - unTarGz /tmp/570237492477157912/geckodriver-v0.29.1-macos-aarch64.tar.gz
2951 [main] TRACE io.github.bonigarcia.wdm.online.Downloader - Deleting file /tmp/570237492477157912/geckodriver-v0.29.1-macos-aarch64.tar.gz
2952 [main] TRACE io.github.bonigarcia.wdm.WebDriverManager - Found driver in post-download: /tmp/570237492477157912/geckodriver
2956 [main] TRACE io.github.bonigarcia.wdm.online.Downloader - Deleting folder /tmp/570237492477157912
2957 [main] TRACE io.github.bonigarcia.wdm.online.Downloader - Driver after extraction ~/.cache/selenium/geckodriver/mac64/0.29.1/geckodriver
2958 [main] INFO  io.github.bonigarcia.wdm.WebDriverManager - Exporting webdriver.gecko.driver as ~/.cache/selenium/geckodriver/mac64/0.29.1/geckodriver
```

I can provide a pull request, let me know how should this be fixed (introduce a new arch?).

**Comment by bonigarcia** [comment]
I have just committed a patch to support Aarch64, it will be available in next version of wdm.

One question @thc202. Why do you need geckodriver 0.29.1? I mean, version 0.29.0 is not valid for you? I'm asking because I'm not sure if geckodriver 0.29.1 should be the default version for latest versions of Firefox in wdm.

**Comment by thc202** [comment]
We (@zaproxy) try to provide the latest versions of the WebDrivers to ensure (or try to) that the browsers can be used, we specify the versions ourselves.

**Comment by psiinon** [comment]
ZAP has an option to launch browsers via selenium. We've found that if we dont provide the very latest webdrivers then things tend to break. Usually it means we cant launch specific browsers.

**Comment by edumco** [comment]
@psiinon and @bonigarcia  this could be solved using flags?

latest    - The very latest driver
stable   - The default version
lts         - The version that is recommended to Firefox ESR (extend support release)

I don't know if other browsers have a LTS version

**Comment by bonigarcia** [comment]
The architecture ARM64 is included in version 4.4.3, just released. This problem should be fixed using that version.
